### ISSUE TITLE
password blanking broken on RHEL8

### ISSUE THREAD

**Issue by jsvob** [RedHatInsights/insights-client#78]
Expected contents of /etc/rhsm/rhsm.conf in insights archive (works correctly on RHEL7.5):

```
blablabla...

# password ******** ********
proxy_password =


blablabla...
```

Actual contents of /etc/rhsm/rhsm.conf in insights archive in RHEL8:

```
blablabla...

# password ******** ********
proxy_password ********

blablabla...
```

This renders the .conf file an invalid ini file, which in turn makes the parser fail.

Versions used in the experiment:

RHEL7.5 (works as expected):
insights-client --version
Client: 3.0.3-9
Core: 3.0.72-1

RHEL8 (doesn't work as expected):
insights-client --version
Client: 3.0.5-1
Core: 3.0.72-1

**Comment by jsvob** [comment]
Any progress?

**Comment by lphiri** [comment]
@jsvob - suggest opening a  JIRA ticket for internal issues.  Also this looks like an insights-core issue.

**Comment by jsvob** [comment]
@lphiri Where shall I file it, please? RHIPLAT1? I see no JIRA project specifically for the client or the core.

Also, I think that it's actually in the client, since the only thing necessary to replicate it is insights-client - the password blanking is broken in the tar.gz archive.

**Comment by lphiri** [comment]
@jsvob put it in RHCLOUD.  Note that the current client is a thin shell around part of Insights core which does all the collection logic.

**Comment by jsvob** [comment]
@lphiri Filed as https://projects.engineering.redhat.com/browse/RHCLOUD-1871. Thanks.

**Comment by jsvob** [comment]
I've just checked again that the issue still exists

```
[root@jsvoboda vagrant]# insights-client --version
Client: 3.0.5-4
Core: 3.0.112-1
[root@jsvoboda vagrant]# cat /etc/redhat-release 
Red Hat Enterprise Linux release 8.0 (Ootpa)
[root@jsvoboda vagrant]# yum upgrade insights-client
Updating Subscription Management repositories.
Last metadata expiration check: 0:03:54 ago on Thu 18 Jul 2019 08:49:53 EDT.
Dependencies resolved.
Nothing to do.
Complete!
[root@jsvoboda vagrant]# yum list insights-client
Updating Subscription Management repositories.
Last metadata expiration check: 0:04:09 ago on Thu 18 Jul 2019 08:49:53 EDT.
Installed Packages
insights-client.noarch                         3.0.5-4.el8                          @AppStream
[root@jsvoboda vagrant]# 


```

**Comment by gravitypriest** [comment]
I'm seeing the "broken" behavior on RHEL 7 as well.
```
⇒  cat etc/rhsm/rhsm.conf | grep password 
# password ******** ********
proxy_password ********
```

**Comment by gravitypriest** [comment]
Alright, looks like the cause is a change to our sed file that was made last year and only exists in the RHEL 8 release so far. Working on a fix.

**Comment by gravitypriest** [comment]
@jsvob Please try replacing the contents of `/etc/insights-client/.exp.sed` with the following and see if the issue persists:

```
s/(password[a-zA-Z0-9_]*)(\s*=*\s*|\s*--md5*\s*)([^[:space:]]*)/\1\2********/
s/(password[a-zA-Z0-9_]*)(\s*\*+\s+)(.+)/\1\2********/
```

**Comment by jsvob** [comment]
@gravitypriest It works. The result is 

```
# password ********
proxy_password =********
```

**Comment by jsvob** [comment]
@gravitypriest What's the progress on this? The impact is that at least one security rule is broken on RHEL8 because of it.

**Comment by gravitypriest** [comment]
I have a fix prepared, but with the current method the fix needs to be included in the RPM. We may want to retool it so that the fix goes into the egg update.

**Comment by jsvob** [comment]
Hey @gravitypriest ,

```
$ grep password rhsm.conf 
# password ******** ********
proxy_password =********

```

Insights runs fine when run locally with the produced archive with that `/etc/insights-client/.exp.sed` (and just to make sure I'm looking at the right things, I tried removing the `=` in `rhsm.conf` in the archive and sure it started producing exceptions).

👍

**Comment by jsvob** [comment]
@gravitypriest I double checked that the file in my test VM contains

```
s/(password[a-zA-Z0-9_]*)(\s*\:\s*\"*\s*|\s*\"*\s*=\s*\"\s*|\s*=+\s*|\s*--md5+\s*|\s*)([a-zA-Z0-9_!@#$%^&*()+=/-]*)/\1\2********/
s/(password[a-zA-Z0-9_]*)(\s*\*+\s+)(.+)/\1\2********/

```

which is - as I see - the same as in the current version of your comment.

Thanks for the fix.

**Comment by gravitypriest** [comment]
Fixed in https://github.com/RedHatInsights/insights-client/commit/a284b8c1081c0328130475ca7f053158233124d1

Will be available in insights-client v3.0.7
